src/inausoft.netCLI.Tests/RegexArgumentDeserializerTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using inausoft.netCLI.Deserialization;
+using inausoft.netCLI.Tests.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace inausoft.netCLI.Tests
+{
+    [TestClass]
+    public class RegexArgumentDeserializerTests
+    {
+        [TestMethod]
+        public void RootCommandHandler_RunsProperCommandHandler_ForMultipleOptions()
+        {
+            //Arrange
+            var stringOptionValue = "sampleString";
+            var intOptionValue = 102;
+
+            var args = new string[]
+                            {
+                                "--boolOption",
+                                "--stringOption", stringOptionValue,
+                                "--intOption", intOptionValue.ToString()
+                            };
+
+            var deserializer = new RegexArgumentDeserializer();
+
+            //Act
+            var command = deserializer.Deserialize<Command1>(string.Join(" ", args));
+
+            //Assert
+            Assert.IsTrue(command.BoolOption);
+            Assert.IsNull(command.NotOptionProperty);
```

Comment:
What does `NotOptionProperty` means? Shouldn't it be better to call it  `NotAssigned`?

Reply:
This is to indicate that this property was not marked as `Option` - to actually verify if it does not had it's value assigned. Be free to propose better solution.